src/Encryption.ts

```diff
@@ -0,0 +1,75 @@
+import Store from "./Store";
+import * as Utils from "./Utils";
+import IDecryptor from "./Decryptor";
+
+/* Encryption is constructed from Encryption.xml 
+It currently only works on spec-compliant epub file formats.  
+Encrypted Webpubs are not supported. */
+export default class Encryption {
+  public readonly encryptionUrl: URL;
+  public readonly resources: string[];
+  /* Fetch Package Document (OEBPS package file) or Webpub Manifest (manifest.json)*/
+  public static async getEncryption(
+    encryptionUrl: URL,
+    store?: Store
+  ): Promise<Encryption> {
+    const fetchEncryption = async (): Promise<Encryption> => {
+      const encryption = await window 
+            .fetch(encryptionUrl.href)
+            .then((response) => response.text())
+            .then((str) => 
+              new DOMParser().parseFromString(str, "text/xml")
+            )
+            .then((data) => JSON.stringify(Utils.xmlToJson(data)));
+      
+      if (store) {
+        await store.set("encryption", JSON.stringify(encryption));
+      }
+      return new Encryption(JSON.parse(JSON.stringify(encryption)), encryptionUrl);
+    };
+
+    // Respond immediately with the encryption from the store, if possible.
+    if (store) {
```

Comment:
There's a bug here -- things work the first time a book is loaded but not on refresh.  Comment this if block out until I figure out what's going on